lib/compress/zstd_compress.c

```diff
@@ -559,6 +559,11 @@ ZSTD_bounds ZSTD_cParam_getBounds(ZSTD_cParameter param)
         bounds.upperBound = (int)ZSTD_urm_enableRowMatchFinder;
         return bounds;
 
+    case ZSTD_c_deterministicRefPrefix:
```

Comment:
opened question  : 
flag name : `ZSTD_c_deterministicRefPrefix`
the name describes pretty well the objective, not the how.

In general, this is a good rule to select a name.

However, in this case, I was wondering :
in which case would anyone select "no" to this property ?
"Do you want a deterministic output ?" "No, I'm fine with unpredictable randomness"

I guess, one only answers "no" if there is something in exchange.
In this case, the only answer I could think of would probably be more speed
(and I don't even know how much speed we are talking about, it could be negligible).

Even then, making the effort to benefit from this speed only makes sense if the user consistently organizes layout so that the prefix always stands just before the destination buffer. In which case ... the output is deterministic.
So it feels strange to have to answer "no" to `ZSTD_c_deterministicRefPrefix` in order to benefit from better speed by concatenating output buffer right after dictionary prefix.

So, I was wondering : presuming _someone_ would be interested in better speed, by ensuring that dictionary prefix and output buffer are contiguous, what would be the "right" parameter name that would feel "correct"?

I could imagine a 3-stages parameter : always separated, always contiguous (in which case it must fail if buffers are not contiguous), automatic (hence non deterministic).

Thoughts ?

Reply:
I would be fine to add this, it would be simple to do. But, it does add more complexity to the API, so there should be some gain, like speed.

I guess that means I should measure the speed difference between ext-dict & prefix dictionaries... I will do that, and if there is a significant gain, I can extend the API.

Reply:
I recall measuring something like a 5% speed gain on a subset of the `httparchive` corpus with a 16KB dict, for contiguous vs. non-contiguous.

Reply:
Prefix-mode is ~2.5% faster at level 1, 1.3% faster at level 3, and 2.5% slower at level 5.

I think that loss is not worth the complexity, especially because the user would probably have to copy their buffers to ensure they're contiguous. I will update the comments to explain the tradeoff a bit more.

Reply:
Agreed with the outcome.

I'm surprised though that prefix mode (which I understand as "contiguous buffers") ends up being _slower_ at level 5. The expectation was that it would be at least as fast, if only by a tiny margin, but not slower.

Reply:
> I'm surprised though that prefix mode (which I understand as "contiguous buffers") ends up being slower at level 5. The expectation was that it would be at least as fast, if only by a tiny margin, but not slower.

That was a typo, it is 2.5% faster.